multiversx_sdk/accounts/ledger_account.py

```diff
@@ -0,0 +1,71 @@
+import logging
+
+from multiversx_sdk.core.address import Address
+from multiversx_sdk.core.message import Message
+from multiversx_sdk.core.transaction import Transaction
+from multiversx_sdk.core.transaction_computer import TransactionComputer
+from multiversx_sdk.ledger.ledger_app import LedgerApp
+
+logger = logging.getLogger("ledger_account")
+
+
+class LedgerAccount:
+    def __init__(self, address_index: int = 0) -> None:
+        self.address_index = address_index
+        self.address = self._get_address()
+        self.nonce = 0
+        self.use_hash_signing = True
+
+    def set_address(self, address_index: int = 0):
+        """Sets the working address for the device. Also sets the `address_index` and `address` properties of the class."""
+        app = LedgerApp()
+        app.set_address(address_index)
+        bech32_address = app.get_address(address_index)
+        app.close()
+
+        self.address_index = address_index
+        self.address = Address.new_from_bech32(bech32_address)
+
+    def sign_transaction(self, transaction: Transaction) -> bytes:
+        transaction_computer = TransactionComputer()
+
+        if not transaction_computer.has_options_set_for_hash_signing(transaction):
```

Comment:
Isn't this checked already by the computer (sorry if mistaken)?

Reply:
somehow. here we explicitly check for `options` to be properly set, but the computer mainly check if the `version` is set to `2` if there are any options present.